### PR TITLE
Create workflow for make upgrade

### PR DESCRIPTION
**Pull request by emcfarlane**
This PR introduces a workflow to run `make upgrade`. Currently for testing it can be run locally or as job through the UI, but will be later updated on a scheduled basis or after a dependabots merge. We can set the schedule in the `.github/workflows/create-upgrade-pr.yaml` file with [schdule syntax](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) to match dependabots frequency. The PR was created through testing the script locally. The header and body are as follows:

# Make upgrade

Make sure to review the changes and merge if everything looks good.

### CODE REVIEW

File: .github/workflows/create-upgrade-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout repository code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ steps.app_token.outputs.token }}
+      - name: Set up Git name and email
+        # TODO: replace with bot's name and email
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
+      - name: Create upgrade PR
```
Comment (bufdev) [inline .github/workflows/create-upgrade-pr.yaml:24]: `name: Make upgrade`

File: .github/workflows/create-upgrade-pr.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Create Upgrade PR
```
Comment (bufdev) [inline .github/workflows/create-upgrade-pr.yaml:1]: `name: Make upgrade`

File: createupgradepr.bash [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Ensure the following environment variables are set:
+: ${DATE}
+#: ${GH_TOKEN} # However, if you are already logged in for GitHub CLI locally, you can remove this line when running it locally.
+
+make upgrade
+
+BRANCH="upgrade/${DATE}"
+git switch -C ${BRANCH}
+git add .
+git commit -m "Upgrade dependencies ${DATE}"
```
Comment (bufdev) [inline createupgradepr.bash:17]: `git commit -am "Make upgrade"`

File: createupgradepr.bash [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+DIR="$(CDPATH= cd "$(dirname "${0}")/../../.." && pwd)"
+cd "${DIR}"
+
+# Ensure the following environment variables are set:
+: ${DATE}
+#: ${GH_TOKEN} # However, if you are already logged in for GitHub CLI locally, you can remove this line when running it locally.
+
+make upgrade
+
+BRANCH="upgrade/${DATE}"
```
Comment (bufdev) [inline createupgradepr.bash:14]: `BRANCH="make-upgrade/${DATE}`

File: createupgradepr.bash [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+git switch -C ${BRANCH}
+git add .
+git commit -m "Upgrade dependencies ${DATE}"
+git push --set-upstream origin ${BRANCH}
+PR_URL=$(gh pr create --title "Upgrade dependencies ${DATE}" --body "Make sure to review the changes and merge it if everything looks good." --base main --head ${BRANCH})
```
Comment (bufdev) [inline createupgradepr.bash:19]: `...and merge if everything...

File: createupgradepr.bash [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
Comment (bufdev) [inline createupgradepr.bash:1]: This looks to be a duplicate file.

File: createupgradepr.bash (hunk 4 above)
Comment (bufdev) [reply createupgradepr.bash:14]: Do this on the make/buf/scripts copy.

File: createupgradepr.bash (hunk 3 above)
Comment (bufdev) [reply createupgradepr.bash:17]: Do this on the make/buf/scripts copy.

File: createupgradepr.bash (hunk 5 above)
Comment (bufdev) [reply createupgradepr.bash:19]: Do this on the make/buf/scripts copy.

File: make/buf/scripts/createupgradepr.bash [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
```
Comment (bufdev) [inline make/buf/scripts/createupgradepr.bash:1]: Make sure to apply comments from other copy of this file.

Change file name to `make/buf/scripts/githubactionmakeupgrade.bash`

**Review by bufdev — APPROVED** [review]

File: .github/workflows/make-upgrade.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+name: Make upgrade
+on: [workflow_dispatch]
+# TODO: schedule this to run weekly or after dependabot update merges
```
Comment (bufdev) [inline .github/workflows/make-upgrade.yaml:3]: What is the status of this? I just want this to run every Monday morning ET at 6AM, is that possible?

Comment (emcfarlane) [reply .github/workflows/make-upgrade.yaml:3]: Okay great. Plan was to run it manually the first time, then update with the schedule.

Comment (emcfarlane) [reply .github/workflows/make-upgrade.yaml]: Scheduled for 10am UTC (6am ET).

**Review by bufdev — APPROVED** [review]

**Pull request merged**